nixos/modules/services/misc/redmine.nix

```diff
@@ -12,12 +12,14 @@ let
   databaseYml = pkgs.writeText "database.yml" ''
     production:
       adapter: ${cfg.database.type}
-      database: ${cfg.database.name}
+      database: ${if cfg.database.type == "sqlite3" then "${cfg.stateDir}/database.sqlite3" else cfg.database.name}
+      ${optionalString (cfg.database.type != "sqlite3") ''
       host: ${if (cfg.database.type == "postgresql" && cfg.database.socket != null) then cfg.database.socket else cfg.database.host}
       port: ${toString cfg.database.port}
       username: ${cfg.database.user}
       password: #dbpass#
       ${optionalString (cfg.database.type == "mysql2" && cfg.database.socket != null) "socket: ${cfg.database.socket}"}
+      ''}
```

Comment:
The indentations are wrong now. Any suggestions on how to write this differently?

```
production:
  adapter: mysql2
  database: redmine
  host: localhost
port: 3306
username: redmine
password:
socket: /run/mysqld/mysqld.sock
```

The configuration with SQLite looks good.
```
production:
  adapter: sqlite3
  database: /var/lib/redmine/database.sqlite3
```

Reply:
YAML is a superset of JSON, meaning a YAML can contain JSON.

So you could probably just rewrite it as an attrset using either `builtins.toJSON` or `format.generate` (like `configurationYml` does) instead :)

Reply:
Alright, I will look into that. Thanks for the hint :)

Reply:
We even have `lib.toYAML` which does exactly that ;)